### PR TITLE
updating ESM configuration with identity mapping commands

### PR DESCRIPTION
**Pull request by Shobhajayanna**
Describe your pull request here:

it is to update the ESM configuration with identity mapping commands.

List the file(s) included in this PR:

api-mediation-oidc-authentication.md

After creating the PR, follow the instructions in the comments.

### CODE REVIEW

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md [hunk 1]
```diff
@@ -83,6 +83,28 @@ API ML provides a Zowe CLI plugin to help administrators generate a JCL for crea
...
+IDMAPRN(distributed-identity-registry-name) IDLABEL(label-name)
+
+F ACF2,REBUILD(USR),CLASS(P),DIVISION(IDMAP)
+END
+
+- TSS :
+  TSS ADD(userid) IDMAP(ZWEDNMAP) IDMAPDN('distributed-identity-user-name') -
+  IDMAPRN('distributed-identity-registry-name') IDLABEL('label-name')
+
+  TSS REFRESH
```
Comment (pablocarle) [inline docs/extend/extend-apiml/api-mediation-oidc-authentication.md:106]: See below there are items for RACF, TSS and ACF2 (also remove the CA prefix). These commands should be included in their respective item.

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md [hunk 2]
```diff
@@ -83,6 +83,28 @@ API ML provides a Zowe CLI plugin to help administrators generate a JCL for crea
 
   See the [Identity Federation cli plugin](../../user-guide/cli-idfplugin.md) documentation for details about how to use the plugin tool to set up the mapping in the ESM of your z/OS system.
```
Comment (pablocarle) [inline docs/extend/extend-apiml/api-mediation-oidc-authentication.md:85]: Please rephrase to have first the commands description and then mention the tool as a way to make it easier

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md [hunk 3]
```diff
@@ -79,15 +79,42 @@ A distributed identity consists of two parts:
...
 
-API ML provides a Zowe CLI plugin to help administrators generate a JCL for creating the mapping filter specific for the ESM installed on the target mainframe system.
+Administrators can use the installed ESM functionality to create, delete, list, and query a distributed identity filter/s:
 
-  See the [Identity Federation cli plugin](../../user-guide/cli-idfplugin.md) documentation for details about how to use the plugin tool to set up the mapping in the ESM of your z/OS system.
+  Below are the commands to create distributed identity filter in ESMs.
+
+- For RACF, 
+
+  RACMAP ID(userid) MAP USERDIDFILTER(NAME('distributed-identity-user-name')) REGISTRY(NAME('distributed-identity-registry-name' )) WITHLABEL('label-name')
```
Comment (pablocarle) [inline docs/extend/extend-apiml/api-mediation-oidc-authentication.md:88]: Introduce what those placeholders should be, maybe an example

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md [hunk 4]
```diff
@@ -79,15 +79,42 @@ A distributed identity consists of two parts:
...
+
+- For ACF2, 
+
+  ACF
+  SET PROFILE(USER) DIVISION(IDMAP)
+  INSERT userid.ZWEDNMAP IDMAPDN(distributed-identity-user-name) -
+  IDMAPRN(distributed-identity-registry-name) IDLABEL(label-name)
+
+  F ACF2,REBUILD(USR),CLASS(P),DIVISION(IDMAP)
+  END
```
Comment (pablocarle) [inline docs/extend/extend-apiml/api-mediation-oidc-authentication.md:111]: apply formatting to the code samples

**Review by janan07 — APPROVED** [review]
Reviewed and edited by doc squad

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md [hunk 5]
```diff
@@ -68,26 +68,73 @@ For example, web applications with a secure server side component can use `code
...
 The user identity mapping is defined as a distributed user identity filter, which is maintained by the System Authorization Facility (SAF) / External Security Manager (ESM).
 A distributed identity consists of two parts:
 - A distributed identity name
 - A trusted registry which governs that identity
 
-API ML provides a Zowe CLI plugin to help administrators generate a JCL for creating the mapping filter specific for the ESM installed on the target mainframe system.
+Administrators can use the installed ESM functionality to create, delete, list, and query a distributed identity filter or filters:
 
-  See the [Identity Federation cli plugin](../../user-guide/cli-idfplugin.md) documentation for details about how to use the plugin tool to set up the mapping in the ESM of your z/OS system.
+Use the commands specific to your ESM to create distributed identity filter.
```
Comment (pablocarle) [inline docs/extend/extend-apiml/api-mediation-oidc-authentication.md:84]: What is a distributed identity filter? why not mapping?

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md [hunk 6]
```diff
@@ -68,26 +68,81 @@ For example, web applications with a secure server side component can use `code
...
+  For more details about mapping a distributed user to a logonid, see [IDMAP User Profile Data Records](https://techdocs.broadcom.com/us/en/ca-mainframe-software/security/ca-acf2-for-z-os/16-0/administrating/administer-records/user-profile-records/idmap-user-profile-records.html).
+
+#### Parameters in the ESM commands
+
+- **`userid`**  
+   Specifies the ESM user id
+- **`distributed-identity-user-name`**  
+   Specifies the user id for distributed-identity-registry
+- **`distributed-identity-registry-name`**  
+   Specifies the hostname of the registry
```
Comment (pablocarle) [inline docs/extend/extend-apiml/api-mediation-oidc-authentication.md:133]: is it just the hostname or a full URL?

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md [hunk 7]
```diff
@@ -68,26 +68,81 @@ For example, web applications with a secure server side component can use `code
...
+- **`label-name`**  
+   Specifies the name for the distributed-identity filter
```
Comment (pablocarle) [inline docs/extend/extend-apiml/api-mediation-oidc-authentication.md:135]: Is `filter` the right word?

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md [hunk 8]
```diff
@@ -57,9 +57,9 @@ Ensure that the following prerequisites are met:
 - SAF/ESM is configured with mapping between the mainframe and distributed user identities. For details, see the section [ESM configuration](#esm-configuration) in this topic.
 - ZSS must be enabled and properly configured in the Zowe installation.
```
[the remark this answers is not in the document]
Comment (pablocarle) [reply docs/extend/extend-apiml/api-mediation-oidc-authentication.md:58]: It's from 2.14, enabling of internal mapper is here (default is still ZSS) https://docs.zowe.org/v2.14.x/user-guide/authenticating-with-client-certificates/#enabling-the-internal-api-ml-mapper

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md [hunk 9]
```diff
@@ -55,11 +55,11 @@ Ensure that the following prerequisites are met:
 - Users who require access to mainframe resources using OIDC authentication have a mainframe identity managed by SAF/ESM.
 - Client application users have their distributed identity managed by the OIDC provider. For details, see the section [OIDC provider](#oidc-provider) in this topic.
 - SAF/ESM is configured with mapping between the mainframe and distributed user identities. For details, see the section [ESM configuration](#esm-configuration) in this topic.
-- ZSS must be enabled and properly configured in the Zowe installation.
+- Ensure that the API ML Gateway is configured to use the internal mapper functionality. Alternatively, enable ZSS in the Zowe installation, however using the internal mapper is the recommended method. ZSS is enabled by default.
```
Comment (pablocarle) [inline docs/extend/extend-apiml/api-mediation-oidc-authentication.md:58]: I think this is missing either a link to the internal mapper doc or at least the minimum version of Zowe?

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md [hunk 10]
```diff
@@ -68,26 +68,81 @@ For example, web applications with a secure server side component can use `code
...
+- **For Top Secret:**
+
+```markup
+  TSS ADD(userid) IDMAP(ZWEDNMAP) IDMAPDN('distributed-identity-user-name') - <br>
+  IDMAPRN('distributed-identity-registry-name') IDLABEL('label-name')
+
+  TSS REFRESH
+```
+
+  For more details about mapping a distributed identify username and a distributed registry name to a Top Secret ACID, see [IDMAP Keyword - Implement z/OS Identity Propagation Mapping](https://techdocs.broadcom.com/us/en/ca-mainframe-software/security/ca-top-secret-for-z-os/16-0/administrating/issuing-commands-to-communicate-administrative-requirements/keywords/idmap-keyword-implement-z-os-identity-propagation-mapping.html).
```
Comment (pablocarle) [inline docs/extend/extend-apiml/api-mediation-oidc-authentication.md:110]:
```suggestion
  For more details about mapping a distributed identity username and a distributed registry name to a Top Secret ACID, see [IDMAP Keyword - Implement z/OS Identity Propagation Mapping](https://techdocs.broadcom.com/us/en/ca-mainframe-software/security/ca-top-secret-for-z-os/16-0/administrating/issuing-commands-to-communicate-administrative-requirements/keywords/idmap-keyword-implement-z-os-identity-propagation-mapping.html).
```

File: docs/extend/extend-apiml/api-mediation-oidc-authentication.md (hunk 9 above)
Comment (pablocarle) [reply docs/extend/extend-apiml/api-mediation-oidc-authentication.md:62]: Thank you

**Review by pablocarle — APPROVED** [review]
Thank you for addressing my comments

**Pull request merged**